xls/examples/parameterized/BUILD

```diff
@@ -0,0 +1,99 @@
+load("//xls/build_rules:xls_utilities.bzl", "proto_to_dslx")
+load("@bazel_skylib//rules:common_settings.bzl", "string_flag")
+load(
+    "//xls/build_rules:xls_build_defs.bzl",
+    "xls_dslx_ir",
+    "xls_dslx_library",
+    "xls_ir_opt_ir",
+    "xls_ir_verilog",
+)
+
+# First variant of the design
+proto_to_dslx(
+    name = "parameters_add",
+    proto_schema_path = ":param.proto",
```

Comment:
would we gain anything by pointing to a `proto_library` here? (so that we can get hooked to a `NativeProtoInfo` provider)

Reply:
Using existing bazel constructs to abstract this out looks like a good idea

Reply:
Changed the rule implementation to accept `proto_schema` attribute which provides `ProtoInfo`